libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/ModelOverview.tsx

```diff
@@ -0,0 +1,195 @@
+          text: localization.ModelAssessment.ModelOverview.selectionRate
+        }
+      );
+    } else {
+      // task_type === "regression"
+      selectableMetrics.push(
+        {
+          key: RegressionMetrics.MeanAbsoluteError,
+          text: localization.ModelAssessment.ModelOverview.meanAbsoluteError
+        },
+        {
+          key: RegressionMetrics.MeanSquaredError,
+          text: localization.ModelAssessment.ModelOverview.meanSquaredError
+        },
+        {
+          key: RegressionMetrics.MeanPrediction,
+          text: localization.ModelAssessment.ModelOverview.meanPrediction
+        }
+      );
+    }
+
+    const columns: string[] = [
+      localization.ModelAssessment.ModelOverview.countColumnHeader
+    ];
+    columns.push(
+      ...selectableMetrics
+        .filter((element) =>
+          this.state.selectedMetrics.includes(element.key.toString())
+        )
+        .map((element) => {
+          return element.text;
+        })
+    );
+
+    return (
+      <Stack tokens={{ childrenGap: "10px", padding: "16px 40px 10px 40px" }}>
+        <Text variant="medium">
+          {localization.Interpret.ModelPerformance.helperText}
+        </Text>
+        {!this.props.showNewModelOverviewExperience && <OverallMetricChart />}
+        {this.props.showNewModelOverviewExperience && (
+          <>
+            <Dropdown
+              placeholder={
+                localization.ModelAssessment.ModelOverview
+                  .metricSelectionDropdownPlaceholder
+              }
+              label={localization.ModelAssessment.ModelOverview.metricsDropdown}
+              selectedKeys={this.state.selectedMetrics}
+              options={selectableMetrics}
+              onChange={this.onMetricSelectionChange}
+              multiSelect
+              styles={{ dropdown: { width: 400 } }}
+            />
+            <DatasetCohortStatsTable
+              selectableMetrics={selectableMetrics}
+              selectedMetrics={this.state.selectedMetrics}
+            />
+          </>
+        )}
+      </Stack>
+    );
+  }
+
+  private onMetricSelectionChange = (
+    _: React.FormEvent<HTMLDivElement>,
+    item?: IDropdownOption
+  ): void => {
+    if (item && item.selected !== undefined) {
+      if (
+        item.selected &&
+        !this.state.selectedMetrics.includes(item.key.toString())
+      ) {
+        this.setState({
+          selectedMetrics: this.state.selectedMetrics.concat([
+            item.key.toString()
+          ])
+        });
+      }
+      if (
+        !item.selected &&
+        this.state.selectedMetrics.includes(item.key.toString())
+      ) {
+        const selectedMetrics = this.state.selectedMetrics;
+        const unselectedMetricIndex = selectedMetrics.findIndex(
+          (key) => key === item.key.toString()
+        );
+        // remove unselected metric
+        selectedMetrics.splice(unselectedMetricIndex, 1);
```

Comment:
I was commenting to see if we want to use .filter()? `const newSelectedMetrics = selectedMetrics.filter(key => key !== item.key.toString())`  
After research online, looks like .splice() wins. So no change needed here :D
https://stackoverflow.com/questions/44435141/remove-object-in-array-using-filter-and-splice-which-one-is-best-approach-in-jav